### PR TITLE
feat: command group distinction

### LINKED ISSUE #836: Improve Command Group Distinction
### Improvement Description

Enhance Command Group differentiation by introducing a subtle horizontal line for a more polished and streamlined visual appearance.

### Rationale

The addition of a separation line reduces redundancy and offers a clearer visual hierarchy for improved user comprehension.

dc-enc

### Proposed Solution

Include a thin horizontal line at the bottom of each Command Group to visually separate them. This can be achieved by adding a bottom border to the `CommandGroup` component, creating a more polished and organized appearance.

### Alternatives (optional)

_No response_

### Additional Context

_No response_

### Please check the boxes that apply to this improvement suggestion.

- [X] I have searched the existing issues and improvement suggestions to avoid duplication.
- [X] I have provided a clear description of the improvement being suggested.
- [X] I have explained the rationale behind this improvement.
- [X] I have included any relevant technical details or design suggestions.
- [X] I understand that this is a suggestion and that there is no guarantee of implementation.

### PR DESCRIPTION
**Pull request by Gautam-Hegde**
fixes #836 

- Explicit `div` is used instead of `<CommandSeparator/>` , since it failed to render borders for dynamic search results, but only works for initial menu.

(initial menu)
cgrp

(search results)
dyanmic

### CODE REVIEW

**Comment by Mythie** [comment]
Hey there, it looks like you haven't accepted our contributor license agreement yet. In order for us to accept your pull request we ask that you please fill out the CLA:

https://documen.so/cla

File: packages/ui/primitives/command.tsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ const CommandItem = React.forwardRef<
   <CommandPrimitive.Item
     ref={ref}
     className={cn(
-      'aria-selected:bg-accent aria-selected:text-accent-foreground relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50',
+      'aria-selected:bg-accent aria-selected:text-accent-foreground relative flex cursor-default select-none items-center rounded-xl -my-1 -mx-2 px-2 py-1.5 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50',
```
Comment (Gautam-Hegde) [inline packages/ui/primitives/command.tsx:124]:
```suggestion
      'aria-selected:bg-accent aria-selected:text-accent-foreground relative flex cursor-pointer select-none items-center rounded-xl -my-1 -mx-2 px-2 py-1.5 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50',
```
I think `cursor-pointer` makes more sense here

Comment (Gautam-Hegde) [reply packages/ui/primitives/command.tsx:124]: cgrpui

Comment (Gautam-Hegde) [reply packages/ui/primitives/command.tsx:124]: Theme menu changes preview

theme-menu

**Review by dguyen — APPROVED** [review]
It looks good, thank you :)

**Pull request merged**